pyproject.toml

```diff
@@ -31,6 +31,7 @@ Twisted = "20.3.0"
 channels-redis = "^3.2.0"
 django-clacks = "^0.2.0"
 requests = "^2.26.0"
+django-anymail = {extras = ["amazon_ses", "mailgun", "sendgrid", "console", "mailjet", "mandrill", "postal", "postmark", "sendinblue", "sparkpost"], version = "^8.4"}
```

Comment:
- Does this mean we can remove sendgrid as a pinned dependency?
- Does core *need* all of these extras, or can we simply add our own extra to provide these in a production environment?
- Don't these providers expose endpoints for authenticating via SMTP? If so why are we using anymail over exposing configuration options for SMTP (`EMAIL_HOST`, `EMAIL_PORT`, `EMAIL_HOST_USER` and `EMAIL_HOST_PASSWORD`)?

Reply:
> Does this mean we can remove sendgrid as a pinned dependency?
Yes, I'll do this
> Does core need all of these extras, or can we simply add our own extra to provide these in a production environment?
If we had a way to do optional dependencies for Core, I'd say yes. That's not a thing we have right now however (at least under Docker)
> Don't these providers expose endpoints for authenticating via SMTP?
Some but not all, and even those that do generally consider the SMTP gateway to be a second-class citizen in terms of support. Additionally, we've always allowed API sending in the past and I don't see a reason to stop.